sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -616,10 +621,23 @@ def create_accumulator(self, *args, **kwargs):
     return [c.create_accumulator(*args, **kwargs) for c in self._combiners]
 
   def merge_accumulators(self, accumulators, *args, **kwargs):
-    return [
-        c.merge_accumulators(a, *args, **kwargs) for c,
-        a in zip(self._combiners, zip(*accumulators))
-    ]
+    # Make sure that `accumulators` is an iterator (so that the position is
+    # remembered).
+    accumulators = iter(accumulators)
+    result = next(accumulators)
+    while True:
+      # Load accumulators into memory and merge in batches to decrease peak
+      # memory usage.
+      accumulators_batch = list(
```

Comment:
Just a thought, should the size of this list be a function of the number of accumulators as well?

Reply:
You mean the # of packed accumulators (i.e. tupled CombineFns) ?

Yes, ideally, it's a function of # of packed accumulators and their sizes, but since we don't know their sizes we wouldn't be able to derive a meaningful estimate for the batch size given just the # of packed accumulators. 
It might still make sense to have default have a ~ reverse linear dependency on the # of packed accumulators though, but I'm not certain how to come up with a magic number, e.g. `merge_accumulators_batch_size` = max(100, 10000/# of combiners in a tuple). WDYT?

Reply:
Yes, I was thinking something like that. Your formula looks fine to me. I might even go with `max(10, 1000/# of combiners in a tuple)` just to be a bit more conservative.

Reply:
Done.